app/main.py

```diff
@@ -1,214 +1,56 @@
-        "hp": 75,
-        "armour": [
-            {
-                "part": "helmet",
-                "protection": 15,
-            },
-            {
-                "part": "breastplate",
-                "protection": 20,
-            },
-            {
-                "part": "boots",
-                "protection": 10,
-            }
-        ],
-        "weapon": {
-            "name": "Two-handed Sword",
-            "power": 55,
-        },
-        "potion": None,
-    },
-    "mordred": {
-        "name": "Mordred",
-        "power": 30,
-        "hp": 90,
-        "armour": [
-            {
-                "part": "breastplate",
-                "protection": 15,
-            },
-            {
-                "part": "boots",
-                "protection": 10,
-            }
-        ],
-        "weapon": {
-            "name": "Poisoned Sword",
-            "power": 60,
-        },
-        "potion": {
-            "name": "Berserk",
-            "effect": {
-                "power": +15,
-                "hp": -5,
-                "protection": +10,
-            }
-        }
-    },
-    "red_knight": {
-        "name": "Red Knight",
-        "power": 40,
-        "hp": 70,
-        "armour": [
-            {
-                "part": "breastplate",
-                "protection": 25,
-            }
-        ],
-        "weapon": {
-            "name": "Sword",
-            "power": 45
-        },
-        "potion": {
-            "name": "Blessing",
-            "effect": {
-                "hp": +10,
-                "power": +5,
-            }
-        }
+from app.Knights.knight import Knight
+from app.Staff.armour import Armour
+from app.Staff.potion import Potion
+from app.Staff.weapon import Weapon
+
+
+def knights_creation(config: dict) -> Knight:
+    character = Knight(config["name"], config["power"], config["hp"])
+    armours = []
+    if config["armour"] is not None:
+        for armour in config["armour"]:
+            armours.append(Armour(
+                armour.get("part"),
+                armour.get("protection")
+            ))
+    if config["potion"] is not None:
+        effect = config["potion"]["effect"]
+        potion = Potion(
+            config["potion"]["name"],
+            effect.get("power", 0),
+            effect.get("hp", 0),
+            effect.get("protection", 0)
+        )
+    else:
+        potion = Potion("Unknown")
+    weapon = Weapon(config["weapon"]["name"], config["weapon"]["power"])
+    character.knight_stats_completing(armours, weapon, potion)
+
+    return character
+
+
+def health_update(fighter: Knight) -> None:
+    if fighter.health_points < 0:
+        fighter.health_points = 0
+
+
+def battle_simulation(fighter1: Knight, fighter2: Knight) -> None:
+    fighter1.health_points -= fighter2.power - fighter1.protection
+    fighter2.health_points -= fighter1.power - fighter2.protection
+    for fighter in [fighter1, fighter2]:
+        health_update(fighter)
```

Comment:
The `battle_simulation` function correctly updates the health points of two fighters and uses the `health_update` function to ensure health points are not negative. This is a good use of the DRY principle.